src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java

```diff
@@ -257,11 +256,31 @@ private EatingState eat()
 
         final Food itemFood = stack.getItem().getFood();
 
+        final Item containerItem = stack.getItem().getContainerItem();
+
         final double satIncrease = itemFood.getHealing() * (1.0 + citizen.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(SATURATION));
 
         citizenData.increaseSaturation(satIncrease / 2.0);
         citizenData.getInventory().extractItem(foodSlot, 1, false);
 
+        if (containerItem != null)
```

Comment:
might double check if it's air too, just to be sure =D

Reply:
Done 👍